### ISSUE TITLE
Version 1.4.1 breaks in dashboards with "Ad hoc filter" type of variables.

### ISSUE THREAD

**Issue by kazeborja** [michaeldmoore/michaeldmoore-multistat-panel#54]
Hi there.

I'm using Grafana 7.1.3 and recently updated the plugin to the last version "1.4.1".

Last version of the plugin seems to break when the dashboard configures an "Ad hoc filter" in the variables.

Seems to be triggered by this piece of code:
    https://github.com/michaeldmoore/michaeldmoore-multistat-panel/blob/master/src/module.js#L124

And the fact that Ad Hoc Filters doesn't seem to have "current", so it fails trying to access the "value" property of undefined.

`Plugin component error TypeError: Cannot read property 'value' of undefined
    at eval (module.js:125)
    at Array.forEach (<anonymous>)
    at MultistatPanelCtrl (module.js:124)
    at Object.invoke (angular.js:5106)
    at angular.js:11076
    at it (angular.js:9926)
    at angular.js:9248
    at angular.js:9113
    at plugin_component.ts:219
    at f (plugin_component.ts:254)`

I don't know much more about Grafana internals, so sorry for not being of more help here.

Let me know in case I can help providing more details.

**Comment by michaeldmoore** [comment]
No need to apologize - this is an excellent catch.  I'll fix it in the
soon-to-be update along with some new features that people have been
clamering for.
Thank you
Michael Moore

**Comment by NikolayTsvetkov** [comment]
Hi @michaeldmoore ,
I'm suffering from this issue and would like to know if you foresee releasing new version of the plugin solving it soon ?
Thank you very much in advance !

Cheers,
Nikolay

**Comment by michaeldmoore** [comment]
Ah - good thing you reminded me.  I will look at this tonight.

**Comment by michaeldmoore** [comment]
OK, After looking through the code, and scratching my head a bit - I don't
think I ever did anything with dashboard variables other than list them to
the debug console.  I was trying to see how this feature could be used, as
the documentation is pretty opaque.  I can see, though, how it breaks whe
ad-hoc dashboard variables are defined, which is what you've been trying to
explain to me all this time.  Sorry it took so long.

I'm getting ready to release a significant update to Multistat - adding
multi-bar per label capability.  I still need to update the readme to
describe the new feature, and also - maybe - add a stacked bar mode which
is something a number of users have requested.

In the meantime, you can grab the current version (1.6.0) directly from
github (I've commented out the variable listing so this shouldn't break
with ad-hoc variables anymore).  Here :
https://github.com/michaeldmoore/michaeldmoore-annunciator-panel

Let me know if you need more detailed instructions.


I won't be releasing the official new version of Multistat (now, I'll have
to call this 1.6.1 at least) for a week or two, then it will take another
couple of weeks before Grafana gets around to approving it and posting it
to the official plugin repository.
Good luck

**Comment by NikolayTsvetkov** [comment]
Hi @michaeldmoore ,

Thank you very much for your quick fix of the problem !
Do you have any news about the new release? You mentioned it should be done in a week or two and I was wondering if it's going to make it in the next Grafana release.

Cheers,
Nikolay

**Comment by michaeldmoore** [comment]
I've been dragging my feet with updating the readme file - everything else
is ready.

Don't worry though, the plugins are completely separated with Grafana
releases themselves.  Grafana staff do a review of new and changed plugins
before adding them to the plugin repository, and generally, this process
slows down whenever they get ready to release a new version of Grafana
itself.  Nevertheless, you should be able to continue using whatever you
have working now and it will continue to work with the next version of
Grafana (cross fingers!)

**Comment by dguendisch** [comment]
Wanted to upgrade Grafana from 7.0.x to 7.2.1 but noticed that my multistat panels remained empty. After seeing it working in new dashboards I eventually realized that the issue on my existing dashboards is ad hoc filters.
Looking forward to multistat 1.6.1 :)

**Comment by NikolayTsvetkov** [comment]
Hi @michaeldmoore ,
Thank you very much for your efforts and the great plugin you are providing.
I'm sorry for bothering you again but do you have any news regarding the new 1.6.1 release ? I have been waiting for it and it would be great if we can get it soon.
Cheers !

**Comment by michaeldmoore** [comment]
Yes, I haven't forgotten, but life does get in the way sometimes.  I still
have a couple of tiny things to do, the major thing being to update the
readme file.  hang in there, and thanks for the supportive words.  My
current forecast is releasing it for review sometime this coming week.
Michael

**Comment by kazeborja** [comment]
Thanks for all the updates Michael!
In the meanwhile I've been growing a bit my knowledge of Grafana internals, although my understanding is the update is ready and only missing some README enhancements.

If there is something where I can lend you a hand with in order to push that last bit to release the new version, please let me now. Otherwise I keep waiting ansious for the new release (which I understand is already in review?)!

**Comment by michaeldmoore** [comment]
3 answers.
Yes, I don't think so and No.

Yes, pretty much the only thinking holding it up is my getting off my
backside and writing a proper readme.
I can't see how anyone can do this for me, thanks though for the offer, I
really appreciate it.
And (red face here) - I haven't *actually *submitted it for approval just
yet - I *have* to do the readme first, there is no short cut here, I'm
afraid.

It's not just a matter of my idleness, I've been very busy with paid work -
part of which makes use of the new enhancements.

Hold on though, I will get to this as soon as I possibly can.

Michael Moore

**Comment by NikolayTsvetkov** [comment]
Hi Michael,

Happy New Year ! 
Is there any chance to have the new version released soon ? 
Thanks !

**Comment by michaeldmoore** [comment]
I really hope so.  Right now, there is a problem with Grafana's new plugin
signing mechanism - it just doesn't work for community plugins with new
toolkit libraries.  Last I heard, Marcus (Grafana big gun) is working on
how to fix this problem.  Grafana 7.3+ allows unsigned plugins, but adds a
horrid-looking 'unsigned' warning if the signing process hasn't been done.

They told me, I'd get better support if I paid for a monthly support
contract - but I can't afford that, so we'll just have to wait.  This last
part, I find pretty crummy as I've donated hundreds and hundreds of hours
in creating and maintaining my plugins for the public good, which
benefits Grafana and now they want me to pay them to keep my plugins
working.  :(

Maybe they will come to their senses.

Michael

**Comment by NikolayTsvetkov** [comment]
Hi Michael,

Thank you once again for your efforts and the provided update around this topic.
What you explained doesn't really sound fair to the contributors spending their free time for the public good, as in your case.
I hope Grafana will understand it sooner than later and the problem will be resolved soon.
Cheers !

**Comment by NikolayTsvetkov** [comment]
Hi Michael,

We have upgraded the plugin to the latest version 1.7 that is solving the problem.
Thank you for your efforts and for the new release. 
I believe this issue can be closed.

Cheers!
